lib/excon/connection.rb

```diff
@@ -46,6 +46,7 @@ def proxy=(new_proxy)
     #     @option params [String] :ciphers Only use the specified SSL/TLS cipher suites; use OpenSSL cipher spec format e.g. 'HIGH:!aNULL:!3DES' or 'AES256-SHA:DES-CBC3-SHA'
     #     @option params [String] :proxy Proxy server; e.g. 'http://myproxy.com:8888'
     #     @option params [Fixnum] :retry_limit Set how many times we'll retry a failed request.  (Default 4)
+    #     @option params [Fixnum] :retry_interval Set how long to wait between retries.
```

Comment:
@geemus
>  Also, more generally this reminds me of some similar requests I've seen around allowing exponential backoff with retries. That is often seen as a best practice (vs specifically timed retries), so perhaps we could consider just having that as a boolean option instead of integers? Something to consider anyway, what do you think?

I think that it is a good idea, and I did not implement it only because I did not need it immediately. But I'm happy to expand on this but I'm not quite sure about the right way to do it.

One possibility is to add a :retry_multiplier option which defaults to 1. This multiplier would then be used to grow the internal at each iteration.

I want to preserve the existing behaviour, so this command will still not sleep at all:
```
connection.request(:idempotent => true, :retry_limit => 6)
```

This call will still have constant intervals:
```
connection.request(:idempotent => true, :retry_limit => 6, :retry_internal => 10)
```

And this call will back off with the following intervals 1s, 1.5s, 2.25s,  3.4s, 5.1s, 7.6s:
```
connection.request(:idempotent => true, :retry_limit => 6, :retry_internal => 0.5, :retry_multiplier => 1.5)
``` 

Do you think that would match up with previous requests?